lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  length = list.length
+  if length == 0
+    return []
+  end 
+  (1...length).each do |k|
+    if list[k] == list[k-1]
+      i = k
+      until i > length || list[i-1] == nil do 
+        list[i - 1] = list[i]
+        i += 1
+      end 
+    end
+    k += 1 
+  end 
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n), where n is total length of all strings combined 
```

Comment:
I would say O(n * m) where n is the number of strings and m is the number of letters in the average string.  Unless the strings are small in which case you could say O(n)